cli/azd/pkg/project/service_target_aks.go

```diff
@@ -0,0 +1,177 @@
+package project
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"path/filepath"
+	"time"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/azure"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/environment/azdcontext"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/azcli"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/docker"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/kubectl"
+)
+
+type aksTarget struct {
+	config           *ServiceConfig
+	env              *environment.Environment
+	scope            *environment.TargetResource
+	containerService azcli.ContainerServiceClient
+	az               azcli.AzCli
+	docker           docker.Docker
+	kubectl          kubectl.KubectlCli
+}
+
+func (t *aksTarget) RequiredExternalTools() []tools.ExternalTool {
+	return []tools.ExternalTool{t.docker}
+}
+
+func (t *aksTarget) Deploy(ctx context.Context, azdCtx *azdcontext.AzdContext, path string, progress chan<- string) (ServiceDeploymentResult, error) {
```

Comment:
It would be good to have a command that builds and pushes images and another that applies the manifest.  

We have a discussion on azd build happening, might make sense to think about doing that with this.

Reply:
Yes, sounds like `azd build` will happen.  Not going to get that in as part of this PR but we'll refactor as needed.